include/illixr/record_logger.hpp

```diff
@@ -160,26 +158,24 @@ class record {
                 spdlog::get("illixr")->error("[record_logger] Caller passed: {}; record_header specifies: {}",
                                              values_[column].type().name(),
                                              record_header_->get().get_column_type(column).name());
-                abort();
+                throw std::runtime_error("[record_logger] Caller got wrong type for column " + std::to_string(column) + " of " +
+                                         record_header_->get().get_name() +
+                                         "\n[record_logger] Caller passed: " + values_[column].type().name() +
+                                         "; record_header specifies: " + record_header_->get().get_column_type(column).name());
             }
         }
-#endif
     }
 
     record() = default;
 
     ~record() {
-#ifndef NDEBUG
         if (record_header_ && !data_use_indicator_.is_used()) {
             spdlog::get("illixr")->error("[record_logger] Record was deleted without being logged.");
-            abort();
```

Comment:
Is there a need to throw another `std::runtime_error` here to ensure consistency?

Reply:
I don't think so because destructors are labeled as `no-throw` and shouldn't throw as a general rule. This particular error is not severe enough to break that mold.